### DISCUSSION TITLE
编辑的时候当输入一些代码的时候无法保存，出现 502

### DISCUSSION THREAD

**Discussion by CoderXiaopang** [halo-dev/halo#6578] [Q&A]
### 系统信息

- 外部访问地址: 
- 启动时间: 2024-08-29 16:11
- 版本: 2.19.0-rc.3
- 构建时间: 2024-08-29 13:59
- Git Commit: 5bd6d91
- Java: IBM Semeru Runtime Open Edition / 21.0.4+7-LTS
- 数据库: H2 / 2.2.224
- 操作系统: Linux / 5.10.120
- 已激活主题: [Hao 1.5.7](https://github.com/chengzhongxue/halo-theme-hao)
- 已启动插件:
  - [AI 助手 1.0.0](https://www.halo.run/store/apps/app-riNgb)
  - [图床插件 1.1.0](https://github.com/liuyiwuqing/halo-plugin-picture-bed)
  - [PostChat 1.0.3](https://github.com/zhheo/halo-plugin-postchat)
  - [blog-hao 1.1.0](https://github.com/wenjing-xin/plugin-blog-hao)
  - [文本绘图 1.3.0](https://github.com/halo-sigs/plugin-text-diagram)
  - [链上好文 1.0.1](https://github.com/guqing/plugin-share-post-via-link)
  - [Sitemap 1.1.2](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件 1.6.2](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.3.0](https://github.com/halo-dev/plugin-feed)
  - [评论组件 2.4.0](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场 1.4.0](https://www.halo.run/store/apps/app-VYJbF)

### 使用的哪种方式运行？

Docker

### 发生了什么？

无法保存
<img width="304" alt="image" src="https://github.com/user-attachments/assets/690d3cec-a4fd-497b-b87d-af0ceceb6680">

### 复现步骤

```
'''python

from PIL import Image
class ImageProcessor:
    def __init__(self):
       pass



    def resize_and_pad(self, img_path=None,img = None,target_size=(512, 512)):
        """
        调整图像大小并填充到目标尺寸。

        Parameters:
            target_size (tuple): 目标尺寸 (width, height)。默认为 (512, 512)。

        Returns:
            Image: 处理后的图像。
        """
        if not img:
            img = Image.open(img_path)

        if img.width > img.height:
            img = img.rotate(90, expand=True)

        aspect_ratio = img.width / img.height
        if aspect_ratio > 1:
            new_width = min(target_size[0], img.width)
            new_height = int(new_width / aspect_ratio)
        else:
            new_height = min(target_size[1], img.height)
            new_width = int(new_height * aspect_ratio)

        img_resized = img.resize((new_width, new_height), Image.LANCZOS)

        new_img = Image.new("RGB", target_size, (255, 255, 255))
        x1 = (target_size[0] - new_width) // 2
        y1 = (target_size[1] - new_height) // 2
        new_img.paste(img_resized, (x1, y1))

        return new_img

    def rotate_90_degrees(self):
        """
        将图像顺时针旋转 90 度四次，并返回四张旋转后的图像。

        Returns:
            list: 包含四张旋转后图像的列表。
        """
        rotated_images = []
        img = self.img
        for _ in range(4):
            img = img.rotate(90, expand=True)
            rotated_images.append(img.copy())
        return rotated_images


if __name__ == '__main__':
    img_processor = ImageProcessor()
    result_img = img_processor.resize_and_pad("/Users/quxiaopang/Desktop/12222222.jpeg")
    result_img.show()

'''
```

**Comment by CoderXiaopang** [comment]
测试代码：

[test.py.zip](https://github.com/user-attachments/files/16796235/test.py.zip)

**Comment by ruibaby** [comment]
未复现此问题，建议检查是否是 Nginx 或者防火墙拦截。

<img width="1172" alt="image" src="https://github.com/user-attachments/assets/5cf212bf-4a0e-4bec-986e-b9acf7e846cb">

/triage needs-information

**Comment by CoderXiaopang** [comment]
我是 docker 内部署的，试过很多版本 都这样  好难受。。

**Comment by ruibaby** [comment]
上面已经说了，先检查 Nginx 或者你前置的防火墙。

**Comment by CoderXiaopang** [comment]
> 上面已经说了，先检查 Nginx 或者你前置的防火墙。

和防火墙有什么关系啊，现在是有一些能保存  有一些不可以   从 markdown 导入的一些复杂的长文档就不行，简单写几句可以保持

**Comment by ruibaby** [comment]
> > 上面已经说了，先检查 Nginx 或者你前置的防火墙。
> 
> 和防火墙有什么关系啊，现在是有一些能保存 有一些不可以 从 markdown 导入的一些复杂的长文档就不行，简单写几句可以保持

那你倒是提供日志啊，什么都不提供谁知道是什么问题。

**Comment by JohnNiang** [comment]
同样无法复现以上问题，无论是在线上还是本地。

- 如果有 CDN，建议排查一下配置是否正确。
- 如果使用了反向代理工具，建议排查一下配置是否正确。
